### PR TITLE
Explainer and spec update for header errors debug reports

### LINKED ISSUE #801: Debug reports for parse failures
This came up in the May 15 2023 call, especially with respect to https://github.com/WICG/attribution-reporting-api/issues/793 which introduces more parse failures. See also https://github.com/WICG/attribution-reporting-api/issues/525 which discusses this among other things.

There is a problem with how debug reports are designed which requires that we successfully parse the registration JSON before we can learn whether the caller opted into debug reports. It might be worthwhile to rethink this opt-in API surface if we want to support sending debug reports on parsing failures.

One idea we discussed briefly on the call was to signal the opt-in "out of band" e.g. in a separate response header:
```http
Attribution-Reporting-Debug: ?1
Attribution-Reporting-Register-Source: ...
```

The API could process the `Debug` header first and allow for signaling errors regarding other ARA headers.

### PR DESCRIPTION
**Pull request by linnan-github**
Fixes #801

***
<a href="https://pr-preview.s3.amazonaws.com/linnan-github/conversion-measurement-api/pull/1180.html" title="Last updated on Feb 29, 2024, 2:07 PM UTC (39c49e3)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/1180/6c86af5...linnan-github:39c49e3.html" title="Last updated on Feb 29, 2024, 2:07 PM UTC (39c49e3)">Diff</a>

### CODE REVIEW

**Comment by linnan-github** [comment]
@arpanah Could you PTAL, thanks!

**Review by apasel422 — DISMISSED** [review]

File: EVENT.md [hunk 1]
```diff
@@ -824,6 +825,41 @@ debug reports without cross-site leak.
 TODO: Consider supporting debug reports for attribution registrations inside a
 fenced frame tree.
 
+### Optional: header errors debugging reports
+
+The headers related to the Attribution Reporting API can be validated by the [header
+validator](https://github.com/WICG/attribution-reporting-api/tree/main/ts#header-validator).
+In order to better facilitate API debugging, we also allow developers to
+monitor failures in the header validation.
```
[the remark this answers is not in the document]
Comment (linnan-github) [reply EVENT.md]: Updated, thanks.

File: index.bs [hunk 2]
```diff
@@ -1846,14 +1898,35 @@ To <dfn export>process an attribution eligible response</dfn> given a [=suitable
...
+    : "`value`"
+    :: |headerValue|.
+1. Let |data| be a new [=attribution debug data=] with the items:
+    : [=attribution debug data/data type=]
+    :: "<code>[=header errors debug data type/header-parsing-error=]</code>"
+    : [=attribution debug data/body=]
+    :: |body|
+1. Run [=obtain and deliver a debug report=] with « |data| » and |reportingOrigin|.
+
+Note: The user agent may optionally set the description of the error in |body|["`error`"].
```
[the remark this answers is not in the document]
Comment (linnan-github) [reply index.bs]: Updated to mention that it could be any type, thanks.

**Review by apasel422 — APPROVED** [review]

**Review by akashnadan — APPROVED** [review]

File: EVENT.md [hunk 3]
```diff
@@ -824,6 +825,41 @@ debug reports without cross-site leak.
 TODO: Consider supporting debug reports for attribution registrations inside a
 fenced frame tree.
 
+### Optional: header errors debugging reports
```
Comment (arpanah) [inline EVENT.md]: nit: errors->error

File: EVENT.md [hunk 4]
```diff
@@ -40,6 +40,7 @@ extension on top of this.
     - [Attribution-success debugging reports](#attribution-success-debugging-reports)
     - [Verbose debugging reports](#verbose-debugging-reports)
       - [Reporting endpoints](#reporting-endpoints)
+  - [Optional: header errors debugging reports](#optional-header-errors-debugging-reports)
```
Comment (arpanah) [inline EVENT.md]: nit: errors->error

**Review by arpanah — APPROVED** [review]

File: EVENT.md (hunk 3 above)
Comment (linnan-github) [reply EVENT.md]: Already changed to header-error debugging reports.

**Pull request merged**